### DISCUSSION TITLE
Irregular embedding responses cause issues with vector DBs

### DISCUSSION THREAD

**Discussion by nigel-daniels** [withcatai/node-llama-cpp#121] [General]
I've been investigating an issue raised on the Langchain.js embedding module that the embeddings return irregular length vectors and this causes issues when trying to use these with vector databases. The expectation is that the resulting embeddings are a set length (for example Ollma returns embeddings of 5120). I tested this out with the following code:

```
import { LlamaModel, LlamaContext, LlamaChatSession } from "node-llama-cpp";
import { OllamaEmbeddings } from "langchain/embeddings/ollama";

const llamaPath = "/your/path/to/ai/llama-2-13b-chat-q8_0.gguf";

const model = new LlamaModel({
  modelPath: llamaPath,
});

const context = new LlamaContext({ model: model });

const val1 = context.encode("hello nice world");
const val2 = context.encode("Hello world");
const val3 = context.encode("Bye bye");

console.log('node-llama-cpp');
console.log('embedding 1: ' + val1.length);
console.log('embedding 2: ' + val2.length);
console.log('embedding 3: ' + val3.length);

const embeddings = new OllamaEmbeddings({
  model: "llama2",
  baseUrl: "http://localhost:11434", // default value
});

const documents = ["hello nice world", "Hello world", "Bye bye"];

const documentEmbeddings = await embeddings.embedDocuments(documents);

console.log('\nollama');
console.log('embedding 1: ' + documentEmbeddings[0].length);
console.log('embedding 2: ' + documentEmbeddings[1].length);
console.log('embedding 3: ' + documentEmbeddings[2].length);
```

I was not sure if this should be raised as an issue or if this would be best handled as an enhancement to set the vector length?  I could pad the results within the Langchain.js but that would create an inconsistency.

**Comment by nigel-daniels** [comment]
BTW the issue raised on Langchain.JS is [#3626](https://github.com/langchain-ai/langchainjs/issues/3626)

**Comment by giladgd** [comment]
@nigel-daniels I haven't added support for embedding yet, but it is planned for the [next version](https://github.com/withcatai/node-llama-cpp/pull/105).

The code [here](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-community/src/embeddings/llama_cpp.ts) seems wrong

**Reply by giladgd** [reply]
I've opened #123, so let's continue there
